coffee-backend/src/main/kotlin/chat/letscoffee/util/CookieUtils.kt

```diff
@@ -0,0 +1,56 @@
+package chat.letscoffee.util
+
+import org.springframework.util.SerializationUtils
+import java.util.*
+import javax.servlet.http.Cookie
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+
+class CookieUtils {
+
+    companion object {
+        fun getCookie(request: HttpServletRequest, name: String): Optional<Cookie> {
+            val cookies = request.cookies
+            if (cookies != null && cookies.size > 0) {
+                for (cookie in cookies) {
+                    if (cookie.name == name) {
+                        return Optional.of(cookie)
+                    }
+                }
+            }
+            return Optional.empty()
+        }
+
+        fun addCookie(response: HttpServletResponse, name: String?, value: String?, maxAge: Int) {
+            val cookie = Cookie(name, value)
+            cookie.path = "/"
+            cookie.isHttpOnly = true
+            cookie.maxAge = maxAge
+            response.addCookie(cookie)
+        }
+
+        fun deleteCookie(request: HttpServletRequest, response: HttpServletResponse, name: String) {
+            val cookies = request.cookies
+            if (cookies != null && cookies.size > 0) {
+                for (cookie in cookies) {
+                    if (cookie.name == name) {
+                        cookie.value = ""
+                        cookie.path = "/"
+                        cookie.maxAge = 0
+                        response.addCookie(cookie)
+                    }
+                }
+            }
+        }
+
+        fun serialize(`object`: Any?): String {
```

Comment:
Naming things `object` in Kotlin is discouraged. Currently it serializes a `OAuth2AuthorizationRequest?`, so we should use that type. Is it supposed to serialize other classes at a later point?